pkg/test/framework/components/apps/local/apps.go

```diff
@@ -203,5 +203,6 @@ func (m *appsImpl) logEnvoyConfig() {
 }
 
 func configDumpStr(a environment.DeployedApp) (string, error) {
-	return envoy.GetConfigDumpStr(a.(*app).agent.GetAdminPort())
+	dump, err := envoy.GetConfigDump(a.(*app).agent.GetAdminPort())
+	return dump.String(), err
```

Comment:
Does calling `String()` on the proto generate json output?

Reply:
Then the call isn't redundant for now. The `configDumpStr` returns the raw json from calling `/cluster_dump`, which is not the same as calling `dump.String()`.

Reply:
Yeah, I think returns the raw json through GetConfigDumpStr is more convenient.
How about make these lines as a func like to avoid repeated codes:
```
       func doConfigDumpGet(){
               requestURL := fmt.Sprintf("http://127.0.0.1:%d/config_dump", adminPort)
               buffer, err := doHTTPGet(requestURL)
	       if err != nil {
		   return "", err
	       }
        }
```
Then when need json, we convert it to json, or convert it to proto.

Reply:
Sure that's fine.

Reply:
ok, will fix it.